storm-core/test/jvm/org/apache/storm/serialization/SerializationTest.java

```diff
@@ -0,0 +1,160 @@
+        return serializer.serialize(vals);
+    }
+
+    public List deserialize(byte[] bytes, Map conf) throws IOException {
+        KryoValuesDeserializer deserializer = new KryoValuesDeserializer(mkConf(conf));
+        return deserializer.deserialize(bytes);
+    }
+
+    public List roundtrip(List vals) {
+        return roundtrip(vals, new HashMap());
+    }
+
+    public List roundtrip(List vals, Map conf) {
+        List ret = null;
+        try {
+            ret = deserialize(serialize(vals, conf), conf);
+        } catch (IOException e) {
+            LOG.error("Exception when serialize/deserialize ", e);
+        }
+        return ret;
+    }
+
+    @Test
+    public void testJavaSerialization() {
+        Object obj = new TestSerObject(1, 2);
+        List<Object> vals = Lists.newArrayList(obj);
+
+        Map<String, Object> conf = new HashMap<>();
+        conf.put(Config.TOPOLOGY_KRYO_REGISTER, new HashMap<String, String>() {{
+            put("org.apache.storm.testing.TestSerObject", null);
+        }});
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, false);
+        try {
+            roundtrip(vals, conf);
+            Assert.fail("Expected Exception not Thrown for config: " + conf);
+        } catch (Exception e) {
+        }
+
+        conf.clear();
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, true);
+        Assert.assertEquals(vals, roundtrip(vals, conf));
+    }
+
+    @Test
+    public void testKryoDecorator() {
+        Object obj = new TestSerObject(1, 2);
+        List<Object> vals = Lists.newArrayList(obj);
+
+        Map<String, Object> conf = new HashMap<>();
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, false);
+        try {
+            roundtrip(vals, conf);
+            Assert.fail("Expected Exception not Thrown for config: " + conf);
+        } catch (Exception e) {
+        }
+
+        conf.put(Config.TOPOLOGY_KRYO_DECORATORS, Lists.newArrayList("org.apache.storm.testing.TestKryoDecorator"));
+        Assert.assertEquals(vals, roundtrip(vals, conf));
+    }
+
+
+    public static String mkString(int size) {
+        StringBuilder sb = new StringBuilder();
+        while (size-- > 0) {
+            sb.append("a");
+        }
+        return sb.toString();
+    }
+
+    public void isRoundtrip(List vals) {
+        Assert.assertEquals(vals, roundtrip(vals));
+    }
+
+    @Test
+    public void testStringSerialization() {
+        isRoundtrip(Lists.newArrayList("a", "bb", "cbe"));
+        isRoundtrip(Lists.newArrayList(mkString(64 * 1024)));
+        isRoundtrip(Lists.newArrayList(mkString(1024 * 1024)));
+        isRoundtrip(Lists.newArrayList(mkString(1024 * 1024 * 2)));
+    }
+
+    /** when we move to java totally, we can remove this test function entirely **/
+    @Test
+    public void testClojureSerialization() {
+        isRoundtrip(PersistentVector.create(RT.keyword(null, "a")));
```

Comment:
I tried out running tests with test-clojure-serialization remaining in clojure file and calling SerializationTest. All the tests passed. You just need to make isRoundTrip in SerializationTest public. You can see the changes here. Let me know if you are not able to open the link. 
https://github.com/apache/storm/compare/master...abhishekagarwal87:STORM-1250?expand=1